src/App.js

```diff
@@ -1,25 +1,27 @@
 import './App.css';
-import Homepage from './pages/Homepage';
-import Wishlist from './pages/Wishlist';
-import ProductListingPage from './pages/ProductListingPage';
-import Cart from './pages/Cart';
+import {
+  Homepage,
+  Wishlist,
+  ProductDetailPage,
+  ProductListingPage,
+  Cart,
+} from './pages';
+
 import { AuthModal } from './components';
-import { useAuthModal } from './contexts/auth-modal-context';
+import { useAuthModal, useAuth } from './hooks';
 import { Routes, Route, Navigate } from 'react-router-dom';
-import ProductDetailPage from './pages/ProductDetailPage';
 
 function App() {
   const { isAuthModal, resetModal, switchModal, isAuthTypeLogin } =
     useAuthModal();
 
+  const { isAuth } = useAuth();
+
   return (
     <div className="App">
       <Routes>
```

Comment:
Suggestion (you can work on this later) - But you can move this to a separate file and them import the component here/